### PR TITLE
[Color Sytem] Apply color system to Breadcrumbs

### PR DESCRIPTION
**Pull request by dleroux**
addresses: https://github.com/Shopify/polaris-ux/issues/334

## WHAT is this pull request doing?

Apply color system to Breacrumbs

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the [Polaris UI kit](https://polaris.shopify.com/resources/polaris-ui-kit)

### CODE REVIEW

File: src/components/Breadcrumbs/Breadcrumbs.scss [hunk 1]
```diff
@@ -30,16 +30,16 @@ $stacking-order: (
...
 
   &:hover,
   &:active {
-    color: color('ink');
+    color: var(--p-text-on-surface, color('ink'));
     text-decoration: none;
 
     .Icon {
-      @include recolor-icon(color('ink'));
+      @include recolor-icon(var(--p-icon-on-surface, color('ink')));
```
Comment (dleroux) [inline src/components/Breadcrumbs/Breadcrumbs.scss]: I've used `--p-icon` but this causes an issue in dark mode where the icon doesn't match the text color but it does on light mode. What's the approach to fix this inconsistent:

1. change `iconSubduedOnSurface` to match`textSubduedOnSurface` in `color-adjustments.ts`?
2. or just use the `text[Subdued]OnSurface` for both the icon and the text?

File: src/styles/shared/_buttons.scss [hunk 2]
```diff
@@ -288,7 +288,7 @@
 }
 
 @function plain-button-background() {
-  @return rgba(color('ink'), 0.1);
+  @return var(--p-surface-foreground-subdued, rgba(color('ink'), 0.1));
```
Comment (dleroux) [inline src/styles/shared/_buttons.scss]: This is where my [alpha](https://github.com/Shopify/polaris-ux/issues/385) question came from.

Comment (dleroux) [reply src/styles/shared/_buttons.scss]: That said, should this become the focus rectangle since these are all applied on focus to plain buttons?

File: src/components/Breadcrumbs/Breadcrumbs.scss (hunk 1 above)
Comment (tmlayton) [reply src/components/Breadcrumbs/Breadcrumbs.scss]: This is fine, we can adjust the colors later.

File: src/components/Breadcrumbs/Breadcrumbs.scss [hunk 3]
```diff
@@ -73,9 +73,12 @@ $stacking-order: (
 }
 
 .Icon {
-  @include recolor-icon(color('ink', 'lighter'));
-  height: $icon-size;
+  @include recolor-icon(
+    var(--p-icon-subdued-on-surface, color('ink', 'lighter'))
+  );
+  z-index: z-index(content, $stacking-order);
```
Comment (tmlayton) [inline src/components/Breadcrumbs/Breadcrumbs.scss]: Why is this needed?

File: src/styles/shared/_buttons.scss (hunk 2 above)
Comment (tmlayton) [reply src/styles/shared/_buttons.scss]: A couple things here: First to make this match monochromatic links it would be:

```suggestion
  @return var(--p-neutral-action, rgba(color('ink'), 0.1));
```

Second this style should only get applied to the active state, and focus should only get the blue focus ring.

**Comment by tmlayton** [comment]
This component is similar to link, which is more of a medium+ difficulty as it requires turning on and off pseudo elements depending on the theme.

File: src/components/Breadcrumbs/Breadcrumbs.scss (hunk 3 above)
Comment (dleroux) [reply src/components/Breadcrumbs/Breadcrumbs.scss]: Because before we were using transparency, and the icon was seen through it. The stacking order was applied to the content, but was missed on the icon and wasn't visible.

Comment (tmlayton) [reply src/components/Breadcrumbs/Breadcrumbs.scss]: To keep the existing styling unchanged:

```suggestion
  z-index: var(--p-z-index-content, initial);
```

Comment (tmlayton) [reply src/components/Breadcrumbs/Breadcrumbs.scss]: Actually never mind, let’s drop this change for now.

File: src/styles/shared/_buttons.scss (hunk 2 above)
Comment (tmlayton) [reply src/styles/shared/_buttons.scss]: Let’s drop this change for now

File: src/components/Breadcrumbs/Breadcrumbs.scss [hunk 4]
```diff
@@ -21,65 +16,68 @@ $stacking-order: (
...
     .Icon {
-      @include recolor-icon(color('ink'));
+      @include recolor-icon(var(--p-icon-on-surface, color('ink')));
     }
   }
-
+  // stylelint-disable selector-max-specificity
   &:focus {
     outline: none;
+    .ContentWrapper {
```
Comment (dleroux) [inline src/components/Breadcrumbs/Breadcrumbs.scss]: Adding this content wrapper to attach the focus ring to.

File: src/components/Breadcrumbs/Breadcrumbs.scss [hunk 5]
```diff
@@ -21,65 +16,68 @@ $stacking-order: (
...
+    var(--p-icon-subdued-on-surface, color('ink', 'lighter'))
+  );
   width: $icon-size;
-
-  // We set negative top/ bottom margin so the icon never sets
-  // the height of the content. We set negative left margin to
-  // compensate for the inset viewbox and to match the Page
-  // component’s secondary actions
+  height: $icon-size;
   margin: (-0.5 * $icon-size) 0 (-0.5 * $icon-size) rem(-8px);
```
Comment (dleroux) [inline src/components/Breadcrumbs/Breadcrumbs.scss]: This does not look like its needed, probably because I'm using flexbox.

File: src/utilities/theme/color-adjustments.ts [hunk 6]
```diff
@@ -175,7 +175,7 @@ export const colorAdjustments: ColorAdjustments = {
   iconSubduedOnSurface: {
     baseColor: 'onSurface',
     light: {lightness: 59.8},
-    dark: {lightness: 88},
+    dark: {lightness: 62},
```
Comment (dleroux) [inline src/utilities/theme/color-adjustments.ts]: We'll need to review this, for now, I made it the same as the text in dark mode.
